rust_api/src/routes.rs

```diff
@@ -0,0 +1,91 @@
+// futures_util provides common utilities and extension traits
+// for async programming
+use futures_util::StreamExt;
+
+use axum::{
+    extract::{Extension, Path},
+    http::StatusCode,
+    response::{IntoResponse, Response},
+    Json,
+};
+use mongodb::bson::doc;
+use serde_json::json;
+
+use crate::document::User;
+
+pub enum UserError {
+    NotFound,
+    NotCreated,
+    IO,
+}
+
+// Handler for `GET /user`
+pub async fn get_all_users(
+    Extension(db_client): Extension<mongodb::Database>,
+) -> Result<Json<Vec<User>>, UserError> {
+    let users = db_client.collection::<User>("users");
+
+    let mut cursor = users.find(None, None).await?;
+    let mut retrieved_users: Vec<User> = Vec::new();
+    while let Some(user) = cursor.next().await {
+        retrieved_users.push(user?);
+    }
+
+    Ok(retrieved_users.into())
+}
+
+// Handler for `GET /user/:name`
+pub async fn find_user(
```

Comment:
Sry for flueknep, men tænker at vi holder os til den stil du har over med `get_all_users()` - Tænker det giver mening at holde os til REST navnene(get_user, post_user, delete_user, ect.)

Reply:
Done